src/drt/src/io/io.cpp

```diff
@@ -146,7 +146,40 @@ void io::Parser::setInsts(odb::dbBlock* block)
     tmpInst->setOrient(inst->getOrient());
     int numInstTerms = 0;
     tmpInst->setPinAccessIdx(inst->getPinAccessIdx());
+    dbTransform xform = tmpInst->getUpdatedXform();
+    auto grid = tech_->getManufacturingGrid();
     for (auto& uTerm : tmpInst->getMaster()->getTerms()) {
+      if (master->getMasterType().isBlock()) {
```

Comment:
Does this just apply to every master, not just blocks?

Reply:
Following the Github issue, this should only apply to macros

Reply:
But shouldn't any master with off-grid pins cause this problem?

Reply:
True, but the detailed and macro placement should take care of that and never allow off-grid placement. This should take care of misplaced macros edited by MACRO_PLACEMENT_TCL, if I understand correctly.

Reply:
I am just gonna allow this check for all cells. wouldn't hurt anyway.